### PR TITLE
Reader (Iteration 1)

### PR DESCRIPTION
**Pull request by michael**

### CODE REVIEW

File: src/editor/components/TextureContainerEditor.js [hunk 1]
```diff
@@ -8,9 +8,10 @@ export default class TextureContainerEditor extends ContainerEditor {
 
   _renderNode($$, node) {
     let api = this.context.api
+    let disabled = this.props.disabled
     if (!node) throw new Error("'node' is mandatory")
 
-    let props = { node }
+    let props = { node, disabled }
```
Comment (oliver----) [inline src/editor/components/TextureContainerEditor.js]: I will fix this in substance land

Comment (oliver----) [reply src/editor/components/TextureContainerEditor.js]: If you wait some minutes I can push a better version.

Comment (michael) [reply src/editor/components/TextureContainerEditor.js]: please push :)

**Comment by oliver----** [comment]
Yeah.. I am about to do that. Just our yesterdays work needs some 'polishing' ;)

File: src/editor/components/TextureContainerEditor.js (hunk 1 above)
Comment (oliver----) [reply src/editor/components/TextureContainerEditor.js]: Yeah.. I am about to do that. Just our yesterdays work needs some 'polishing' ;)

Comment (michael) [reply src/editor/components/TextureContainerEditor.js]: I'm working on #564 right now... Take your time.

**Comment by oliver----** [comment]
@michael we need to talk about how we want to proceed with the 'master'/'develop' pattern. We had and are still having a situation now, where we want to continue development while other projects a release, which need some patches over time.
We could either use 'master'/'develop', or 'release'/'master'. But we need two tracks.

**Comment by oliver----** [comment]
And then I'd suggest to use the same pattern everywhere.

**Comment by michael** [comment]
@oliver---- yes I understand. I'd prefer for release/master. And master being the default branch, so contributions always target the latest development version.

**Comment by oliver----** [comment]
Ok. I have some things on 'develop' which I will merge in master.
Create a release branch on the commit we used for the last release.
And this branch need to be rebased.
Are you having uncommited changes?

**Comment by michael** [comment]
I have some very experimental stuff, that  I don't want to commit yet. But please go ahead and i will use git stash to bring back my changes.

**Comment by oliver----** [comment]
ok.

**Comment by michael** [comment]
Can you fix the editor so it also provides the api in context? It's currentry broken.

**Comment by oliver----** [comment]
Will do after the rebase.

**Comment by michael** [comment]
@oliver---- I think we also need to make the referenceManager available to the API. I guess then the component implementations could talk to the ReferencesModel which abstracts all that label generation etc. away. 

What do you think?

First thing I could do is creating the instances earlier and passing them to the API. Next we should fix places that use referenceManager etc. directly and instead use the new ReferenceModel API.

**Comment by oliver----** [comment]
Yep. Everything need to go into the API. Like it was a unified model. Of course with all the 'managers' inside to do the work.

**Comment by michael** [comment]
Ok i will try to start this work then. You can forget about the broken editor, I will fix as part of this restructuring.

**Comment by oliver----** [comment]
Ok. But let me finish the rebase. There are some regressions I need to fix.

**Comment by oliver----** [comment]
@michael Now you can start.

**Comment by oliver----** [comment]
I need to fix test coverage reports in the meantime. (using the same toolchain as in stencila/engine et al.)

**Comment by michael** [comment]
Okay. I only have one hour left today, so probably won't finish this today.

**Comment by oliver----** [comment]
Ok

**Comment by oliver----** [comment]
But maybe you could push your stuff, even unfinished. Otherwise I am a bit blocked tonight.

**Comment by michael** [comment]
Sorry saw that too late. But you can move on and push, i will rebase my stuff and also push asap

**Comment by michael** [comment]
@oliver---- I just pushed that change that moves managers into the API. 

It's basically working but somehow the setup of managers within commands seems not right. When you open the editor, and set the cursor to the title, I get this error:

image

It's a lifecycle thing, like on construction of the command manager we don't yet have setup the managers and API. Don't know how to approach this. I think we generally want to get rid of managers inside editorSession... just don't want to kill too much at once... as it's affecting a lot of code.

Could you take a look?

**Comment by oliver----** [comment]
Will take a look.
I will rebase on the latest master.

**Comment by oliver----** [comment]
@michael this is working now. Pls take a look, particularly at https://github.com/substance/texture/pull/545/commits/0e838c16a3c2983b81f275da47e9f2706fab4e88

**Comment by michael** [comment]
Thank you @oliver---- . Makes sense to do the monkey patch for now. And in the long run let's get all this stuff out of editorSession.

**Comment by oliver----** [comment]
Calling it a first iteration.

**Pull request merged**